internal/controller/postgrescluster/cluster.go

```diff
@@ -53,6 +54,32 @@ func (r *Reconciler) reconcileClusterConfigMap(
 	return clusterConfigMap, err
 }
 
+// patroniLogSize attemps to parse the defined log file size, if configured.
+// If a value is set, this enables volume based log storage and triggers the
+// relevant Patroni configuration. If the value given cannot be parsed, the log
+// file size defaults to 25M and a warning event is triggered.
+func (r *Reconciler) patroniLogSize(ctx context.Context, cluster *v1beta1.PostgresCluster) int64 {
+
+	if cluster.Spec.Patroni != nil {
+		if cluster.Spec.Patroni.Log != nil {
+			if cluster.Spec.Patroni.Log.FileSize != nil {
+
+				fileSize, err := resource.ParseQuantity(*cluster.Spec.Patroni.Log.FileSize)
+				sizeInBytes := fileSize.Value()
+
+				if err != nil {
+					r.Recorder.Eventf(cluster, corev1.EventTypeWarning, "InvalidPatroniLogFileSize",
+						"Error when parsing spec.patroni.log.fileSize. File size will default to 25M.")
+
+					sizeInBytes = 25000000
+				}
```

Comment:
⛏️ I'd prefer that we prevent this with validation somehow.

❓ What happens to Patroni if someone picks something tiny, like `.1Ki` ?

Reply:
Per the comment above, it looks like exposing `resource.Quantity` is the better option. I think reworking the current function to force a log file size minimum may make sense. Any thoughts on what size? The Patroni minimum of 25M doesn't feel terrible.

Reply:
Yeah, 25M works for me. We can go lower in the future should the need arise.